File: app/src/main/java/com/example/pam_app/di/Container.java
package com.example.pam_app.di;

import android.content.Context;

import com.example.pam_app.repository.BucketRepository;
import com.example.pam_app.repository.IncomeRepository;
import com.example.pam_app.repository.LanguagesRepository;
import com.example.pam_app.utils.schedulers.SchedulerProvider;

public interface Container {
    Context getApplicationContext();
    SchedulerProvider getSchedulerProvider();
    BucketRepository getBucketRepository();
    IncomeRepository getIncomeRepository();
    LanguagesRepository getLanguageRepository();
}

File: app/src/main/java/com/example/pam_app/di/Container.kt
package com.example.pam_app.di

import android.content.Context
import com.example.pam_app.repository.BucketRepository
import com.example.pam_app.repository.IncomeRepository
import com.example.pam_app.repository.LanguagesRepository
import com.example.pam_app.utils.schedulers.SchedulerProvider
import kotlin.jvm.Synchronized

interface Container {
    val applicationContext: Context?
    val schedulerProvider: SchedulerProvider?
    val bucketRepository: BucketRepository?
    val incomeRepository: IncomeRepository?
    val languageRepository: LanguagesRepository?
}